### PR TITLE
Implement more of the learning service

### PR DESCRIPTION
**Pull request by PaulLockett**
add implementation and some tests for last set of functions in the learning service

### CODE REVIEW

File: src/core/services/images/LearningService.ts [hunk 1]
```diff
@@ -70,23 +71,138 @@ export class ImageLearningService
...
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    if (!response.ok) {
+      // TODO(@PaulLockett): return typed errors
+      const body = await response.text();
+      throw new Error(
+        `Failed to get model by id. ${response.status} ${response.statusText} - ${body}`
```
Comment (atbe) [inline src/core/services/images/LearningService.ts:93]: Wrong error message `failed to get` should be `failed to delete`

File: src/core/services/images/LearningService.ts [hunk 2]
```diff
@@ -70,23 +71,138 @@ export class ImageLearningService
...
-  findModelVersionById(id: string): Promise<void> {
-    throw new Error("Method not implemented.");
+  /**
+   * Fetches a model version by ID. This is useful for checking the status of a training job.
+   *
+   * @param modelId - The ID of the model version to fetch. (uuid)
+   * @param versionId - The ID of the version to fetch. (uuid)
+   * @returns - The model version.
+   */
+  public async findModelVersionById(modelId: string, versionId: string): Promise<ModelVersionEntity> {
```
Comment (atbe) [inline src/core/services/images/LearningService.ts:188]: We shouldn't be returning models like `ModelVersionEntity` as-is to callers - we need to wrap them with a type that we control 

For example, this method uses the `Remix` type https://github.com/leap-api/Leap-Typescript-SDK/blob/c3cef27c8f8f1bf9ac12226ed2d1845f870b6ba1/src/core/services/images/RemixService.ts#LL34C14-L34C19

And notice how the `Remix` type is a wrapper around the generated model https://github.com/leap-api/Leap-Typescript-SDK/blob/master/src/types/schemas/remix/Remix.ts

File: src/core/services/images/LearningService.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { ImageLearningServiceInterface } from "../../../interfaces/images/Learni
 
 import SdkOptions from "../../../types/SdkOptions";
 
+import {ModelVersionEntity, TrainingSampleEntity, UploadSamplesViaUrlDto} from "../../../generated";
```
Comment (atbe) [inline src/core/services/images/LearningService.ts:5]: We should only ever be importing things from the generated OpenAPI code in our schemas.

**Review by atbe — CHANGES_REQUESTED** [review]

File: src/core/services/images/LearningService.ts (hunk 1 above)
Comment (PaulLockett) [reply src/core/services/images/LearningService.ts]: fixed

File: src/core/services/images/LearningService.ts (hunk 2 above)
Comment (PaulLockett) [reply src/core/services/images/LearningService.ts]: I have wrapped them and made extra types to support this

File: src/core/services/images/LearningService.ts (hunk 3 above)
Comment (PaulLockett) [reply src/core/services/images/LearningService.ts]: these imports are removed

**Pull request merged**